FeedTheSnake/FeedTheSnake/SnakeGame.xaml.cs

```diff
@@ -0,0 +1,436 @@
+
+namespace FeedTheSnake
+{
+
+    /// <summary>
+    ///Draws and moves snake and foods on canvas
+    ///in the User Controls
+    /// </summary>
+    public partial class SnakeGame : UserControl, INotifyPropertyChanged
+    {
+        public enum GameState { ONGOING, PAUSED, OVER, PREPEARING }
+        public enum GameLevel { ZERO, FIRST, SECOND, THIRD }
+
+        public event PropertyChangedEventHandler? PropertyChanged;
+
+        #region Fields
+        private double unit;
+        private static Random rnd = new Random();
+
+        private int score;
+        private GameState state;
+
+        private Snake snake;
+        private Polyline? body;
+        private Ellipse? head;
+
+        private FoodFarm farm;
+        private FoodCollection foodStock;
+
+        private List<Obstacle> obstacles;
+        private List<Rectangle> rectangles; 
+        #endregion
+
+        #region Properties
+        /// <summary>
+        /// time for food to expire
+        /// </summary>
+        public int ExpiretionTime
+        {
+            get
+            { return farm.ExpiretionTime; }
+            set
+            {
+                farm.ExpiretionTime = value;
+
+                OnPropertyChanged(nameof(ExpiretionTime));
+            }
+        }
+        /// <summary>
+        /// Game Score
+        /// </summary>
+        public int Score
+        {
+            get
+            {
+                return score;
+            }
+            set
+            {
+                score = value >= 0 ? value : 0;
+                OnPropertyChanged(nameof(Score));
+            }
+        }
+        /// <summary>
+        /// Current game state
+        /// </summary>
+        public GameState State
+        {
+            get
+            {
+                return state;
+            }
+            set
+            {
+                state = value;
+                OnPropertyChanged("State");
+            }
+        }
+        /// <summary>
+        /// Current game level
+        /// </summary>
+        public GameLevel Level { get; set; } 
+        #endregion
+        public SnakeGame()
+        {
+            InitializeComponent();
+            unit = 20;
+            State = GameState.PREPEARING;
+            Level = GameLevel.ZERO;
+            farm = new FoodFarm(unit);
+            snake = new Snake(unit);
+            body = new Polyline();
+            head = new Ellipse();
+            foodStock = new FoodCollection(farm);
+            obstacles = new List<Obstacle>();
+            rectangles = new List<Rectangle>();
+            Score = 0;
+            ExpiretionTime = 5;
+
+            snakeCanvas.Children.Add(body);
+            snakeCanvas.Children.Add(head);
+
+            farm.inspectionTimer.Tick += DrawFood;
+            farm.inspectionTimer.Tick += DeleteExpiredFoods;
+
+            head.MouseLeftButtonDown += (x, y) => { GameStart(); };
+            snake.PropertyChanged += (x, y) => { Score++; };
```

Comment:
I am not sure that is a good idea to increas score on each property change, just use snake lenght

Reply:
Will fix it wit next version